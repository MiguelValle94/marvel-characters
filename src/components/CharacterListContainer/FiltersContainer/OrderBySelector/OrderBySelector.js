import React from 'react';
import PropTypes from 'prop-types';

import './OrderBySelector.scss'

export default function OrderBySelector({ value, onChange }) {
  return (
    <select
      name='orderBy'
      className='form-control order-by-selector'
      onChange={onChange}
      value={value}
    >
      <option value='name'>Name Asc</option>
      <option value='-name'>Name Desc</option>
      <option value='modified'>Modified Asc</option>
      <option value='-modified'>Modified Desc</option>
    </select>
  );
}

OrderBySelector.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};
