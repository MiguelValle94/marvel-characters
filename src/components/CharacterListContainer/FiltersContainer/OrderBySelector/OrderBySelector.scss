@import '../../../../styles/variables.scss';

.order-by-selector {
  margin-bottom: var(--space);
  width: auto;
}
