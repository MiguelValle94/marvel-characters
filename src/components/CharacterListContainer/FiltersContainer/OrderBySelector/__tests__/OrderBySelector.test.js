import React from 'react';
import { mount } from 'enzyme';

import OrderBySelector from '../OrderBySelector';

const setUpMount = (value, handleOnChange) => {
  const component = mount(
    <OrderBySelector value={value} onChange={handleOnChange} />
  );
  return component;
};

describe('OrderBySelector', () => {
  const handleOnChange = jest.fn();

  it('should render component', () => {
    const component = setUpMount('name', handleOnChange);
    expect(component).toMatchSnapshot();
  });

  it('should call onChange callback on change', () => {
    const component = setUpMount('name', handleOnChange);
    const event = { target: { value: '-name' } };
    const select = component.find('select');

    expect(handleOnChange).not.toHaveBeenCalled();
    select.simulate('change', event);

    expect(handleOnChange).toHaveBeenCalledTimes(1);
  });
});
