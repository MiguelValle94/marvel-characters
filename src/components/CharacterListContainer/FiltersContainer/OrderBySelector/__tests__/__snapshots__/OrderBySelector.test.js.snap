// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBySelector should render component 1`] = `
<OrderBySelector
  onChange={[MockFunction]}
  value="name"
>
  <select
    className="form-control"
    name="orderBy"
    onChange={[MockFunction]}
    value="name"
  >
    <option
      value="name"
    >
      Name Asc
    </option>
    <option
      value="-name"
    >
      Name Desc
    </option>
    <option
      value="modified"
    >
      Modified Asc
    </option>
    <option
      value="-modified"
    >
      Modified Desc
    </option>
  </select>
</OrderBySelector>
`;
